src/mlpack/methods/ann/layer/pixel_shuffle_impl.hpp

```diff
@@ -0,0 +1,151 @@
+    batchSize(0),
+    outputHeight(0),
+    outputWidth(0),
+    sizeOut(0),
+    reset(false)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void PixelShuffle<InputDataType, OutputDataType>::Forward(
+  const arma::Mat<eT>& input, arma::Mat<eT>& output)
+{
+  if (!reset)
+  {
+    batchSize = input.n_cols;
+    sizeOut = size / std::pow(upscaleFactor, 2);
+    outputHeight = height * upscaleFactor;
+    outputWidth = width * upscaleFactor;
+    reset = true;
+  }
+  output.zeros(outputHeight * outputWidth * sizeOut, batchSize);
+  for (size_t n = 0; n < batchSize; n++)
+  {
+    arma::mat inputImage = input.col(n);
+    arma::mat outputImage = output.col(n);
+    arma::cube inputTemp(const_cast<arma::mat&>(inputImage).memptr(), height,
+        width, size, false, false);
+    arma::cube outputTemp(const_cast<arma::mat&>(outputImage).memptr(),
+        outputHeight, outputWidth, sizeOut, false, false);
+
+    for (size_t c = 0; c < sizeOut ; c++)
+    {
+      for (size_t h = 0; h < outputHeight; h++)
+      {
+        for (size_t w = 0; w < outputWidth; w++)
+        {
+          size_t height_index = h / upscaleFactor;
+          size_t width_index = w / upscaleFactor;
+          size_t channel_index = (upscaleFactor * (h % upscaleFactor)) +
+              (w % upscaleFactor) + (c * std::pow(upscaleFactor, 2));
+          outputTemp(w, h, c) = inputTemp(width_index, height_index,
+              channel_index);
+        }
+      }
+    }
+    output.col(n) = outputImage;
+  }
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void PixelShuffle<InputDataType, OutputDataType>::Backward(
+    const arma::Mat<eT>& input, const arma::Mat<eT>& gy, arma::Mat<eT>& g)
+{
+  g.zeros(arma::size(input));
+  for (size_t n = 0; n < batchSize; n++)
+  {
+    arma::mat gyImage = gy.col(n);
+    arma::mat gImage = g.col(n);
+    arma::cube gyTemp(const_cast<arma::mat&>(gyImage).memptr(), outputHeight,
+        outputWidth, sizeOut, false, false);
+    arma::cube gTemp(const_cast<arma::mat&>(gImage).memptr(), height, width,
+        size, false, false);
+
+    for (size_t c = 0; c < sizeOut ; c++)
+    {
+      for (size_t h = 0; h < outputHeight; h++)
+      {
+        for (size_t w = 0; w < outputWidth; w++)
+        {
+          size_t height_index = h / upscaleFactor;
+          size_t width_index = w / upscaleFactor;
+          size_t channel_index = (upscaleFactor * (h % upscaleFactor)) +
+              (w % upscaleFactor) + (c * std::pow(upscaleFactor, 2));
+          gTemp(width_index, height_index, channel_index) = gyTemp(w, h, c);
+        }
+      }
+    }
+    g.col(n) = gImage;
+  }
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename Archive>
+void PixelShuffle<InputDataType, OutputDataType>::serialize(
+    Archive& ar,
+    const unsigned int /* version */)
+{
+  ar & BOOST_SERIALIZATION_NVP(delta);
```

Comment:
This has already been discussed, but we're now waiting on the serialization to be updated to use cereal.

Reply:
I have made a continuation to this PR.

Reply:
You can see them at #2833

Reply:
Ah, sorry I missed it!